modules/ROOT/pages/batch-phases.adoc

```diff
@@ -0,0 +1,131 @@
++
+[[batchJobInstanceId]]
+Within the scope of the Batch Job component, Mule exposes the currently processing batch job instance ID through the `batchJobInstanceId` variable. You can use `vars.batchJobInstanceId` to access the identifier of the current instance in any batch processing phase. The auto-generated identifier is a UUID, but you can change it to a different value. For details, see xref:batch-job-instance-id.adoc[].
++
+. The Batch Job component automatically splits the received message payload into records and stores the records in the stepping queue.
++
+The split is "all or nothing," which means that the Batch Job component either successfully generates and queues a record for every item from the input, or the entire event fails with an error type, such as `MULE:EXPRESSION` message, and an error message, such as `"Expecting Array or Object but got String." evaluating expression: "payload".`
++
+In Anypoint Runtime Manager, the queue name is prefixed with `BSQ`, for example:
++
+image:mruntime-batch-bsq.png[Batch step queue transaction log]
++
+. The Batch Job component starts the execution of a batch job instance that processes the loaded records.
+
+[[phase_process]]
+== Process Phase
+
+The Process phase begins after the Load and Dispatch phase finishes loading records to the queue and starts execution of a batch job instance. In this phase, Batch Step components concurrently pull record blocks of the configured size from the queue to process them (the default batch block size is 100).
+
+Within the Batch Step component, processing takes place in parallel at the block level, but records within each block process sequentially by default. Mule uses its auto-tuning capabilities to determine how many threads to use and the level of parallelism to apply (see xref:execution-engine.adoc[Execution Engine]).
+
+During this phase, processors within the components can access and modify records using the Mule variables `payload` and `vars`. The Batch Job component consumes records after all record processing finishes. So any transmission of _processed_ records to an external server or service must take place _within_ the Batch Step or Batch Aggregator components. For Mule variable propagation rules, see xref:batch-processing-concept.adoc#variable_propagation[Variable Propagation].
+
+After processing records in a block, the Batch Step sends the records back to the stepping queue, where the records await by the next Batch Step component. This process continues until every record passes through every Batch Step component within the same Batch Job component. At the end of this phase, all records are consumed. They are not available for further processing by downstream processors in the Mule flow.
+
+Note that changes to records that take place within a Batch Aggregator component _are not propagated_ outside of Batch Aggregator component. So the changes are not visible to subsequent Batch Step components in the flow.
```

Comment:
```suggestion
Note that since changes to records that take place within a Batch Aggregator component _are not propagated_ outside of Batch Aggregator component, changes are not visible to subsequent Batch Step components in the flow.
```

Reply:
Just wondering if this is better as once sentence to main the cause and effect relationship?